src/viewer.js

```diff
@@ -20,22 +20,24 @@
  *
  */
 
-import ViewerComponent from './components/ViewerComponent.vue'
 import { logger } from './helpers/logger.js'
 import { openMimetypesMarkdown, openMimetypesPlainText } from './helpers/mime.js'
 
 __webpack_nonce__ = btoa(OC.requestToken) // eslint-disable-line
 __webpack_public_path__ = OC.linkTo('text', 'js/') // eslint-disable-line
 
-if (typeof OCA.Viewer === 'undefined') {
-	logger.error('Viewer app is not installed')
-} else {
-	OCA.Viewer.registerHandler({
-		id: 'text',
-		mimes: [...openMimetypesMarkdown, ...openMimetypesPlainText],
-		component: ViewerComponent,
-		group: null,
-		theme: 'default',
-		canCompare: true,
-	})
-}
+document.addEventListener('DOMContentLoaded', async () => {
```

Comment:
@skjnldsv I remember this used to be an issue with viewer, but the console log warning in viewer doesn't seem to be there anymore. Can you confirm this is fine to switch?

Reply:
Good memory! We used to watch changes in registration with Viewer. But we stopped.
You need to register handlers and Viewer will wait for the DOMContentLoaded to be fired to process them, so you should not wait for it, [Viewer will](https://github.com/nextcloud/viewer/blob/1ef62842037915f5a82e9301c4dd2e4e94a32056/src/views/Viewer.vue#L508-L528).

It was before we started going for the global libraries and window._nc_xxxx variables which made it possible to register at any point in time :)

Reply:
If you wanna be safe, you might just add a php `addScript` dependency on Viewer? :thinking: 
But I think we mainly covered this already.

```php
Util::addScript('text', 'text-viewer', 'viewer');
```